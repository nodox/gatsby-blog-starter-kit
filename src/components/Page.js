import React from 'react';

export default class Page extends React.Component {
  constructor(props) {
    super(props);
  }

  render() {
    const {
      title,
      desc
    } = this.props;

    return (
      <div id="main">
        <section className="post">
          <header className="major">
            <h1>{title}</h1>
          </header>
          <div className="image main"><img src="images/pic01.jpg" alt=""/></div>
          <p>{desc}</p>
        </section>
      </div>
    );
  }

}
