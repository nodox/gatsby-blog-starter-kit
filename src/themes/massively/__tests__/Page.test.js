import React from 'react';
import renderer from 'react-test-renderer';
import Page from '../components/Page';

describe('<Page />', () => {
  const data = {
    desc: 'Hello World we need a description',
    title: 'Steven Natera',
  }

  const component = renderer.create(
    <Page desc={data.desc} title={data.title} />
  );

  const props = component.getInstance().props;

  it('should render correctly', () => {
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should have the right props', () => {
    expect(props).toHaveProperty('desc');
    expect(props).toHaveProperty('title');
  });

});
