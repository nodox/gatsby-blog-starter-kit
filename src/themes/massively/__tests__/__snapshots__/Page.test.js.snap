// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Page /> should render correctly 1`] = `
<div
  id="main"
>
  <section
    className="post"
  >
    <header
      className="major"
    >
      <h1>
        Steven Natera
      </h1>
    </header>
    <div
      className="image main"
    >
      <img
        alt=""
        src="images/pic01.jpg"
      />
    </div>
    <p>
      Hello World we need a description
    </p>
  </section>
</div>
`;
